### PR TITLE
ENH: Scrape hyphenated ID sequences

### LINKED ISSUE #118: Scrape hyphenated ID sets in `scrape-collection`
As a plugin user, 
I want the `scrape-collection` action to also scrape sets of accession IDs indicated with hyphens, such as `SRX1479846–50` in the publication [10.1038/s41467-019-13036-1](https://www.nature.com/articles/s41467-019-13036-1). 

(🙌🏼 spotted by @lina-kim during review of PR #116)

### PR DESCRIPTION
**Pull request by adamovanja**
This PR introduces scraping of accession ID sequences denoted as `SRX1479846–50` or `SRX1479846–SRX1479850` and closes #118.

**Testing**
Scrape a collection of your choice with publications having hyphenated ID sequences within, such as:
* `ERS5506097 - ERS5506338` in https://doi.org/10.1111/1365-2656.13428
* `SRX1479811–5` in https://doi.org/10.1038/s41467-019-13036-1
* `ERS1865478–ERS1865519` in https://doi.org/10.1093/infdis/jiab398

### CODE REVIEW

File: q2_fondue/scraper.py [hunk 1]
```diff
@@ -328,6 +377,9 @@ def scrape_collection(
         # get text
         try:
             str_text = zot.fulltext_item(attach_key)['content']
+            # remove character frequently placed before soft hyphen, see
```
Comment (misialq) [inline q2_fondue/scraper.py:380]:
```suggestion
            # remove the soft hyphen, see
```

That will remove hyphens themselves, no?

File: q2_fondue/tests/test_scraper.py [hunk 2]
```diff
@@ -226,6 +226,51 @@ def test_find_accession_ids_special_cases_three_commas(self):
...
+
+    def test_find_accession_ids_special_cases_hyphen_two_digits(self):
+        # example inspired by 10.1038/s41467-019-13036-1
+        txt_diff = 'Scott mWGS: PRJEB1479846–50'
+        exp_proj = ['PRJEB1479846', 'PRJEB1479847', 'PRJEB1479848',
+                    'PRJEB1479849', 'PRJEB1479850']
+        obs_proj = _find_accession_ids(txt_diff, 'bioproject')
+        self.assertListEqual(sorted(exp_proj), sorted(obs_proj))
+
+    def test_find_accession_ids_special_cases_hyphen_three_digits(self):
```
Comment (misialq) [inline q2_fondue/tests/test_scraper.py:245]: Could you please add a test for when the ID rolls over 100? For example, `PRJEB1479098–105`?

File: q2_fondue/scraper.py [hunk 3]
```diff
@@ -217,6 +217,50 @@ def _find_special_id(txt: str, pattern: str, split_str: str) -> list:
     return ids
 
 
+def _find_hyphen_sequence(
```
Comment (misialq) [inline q2_fondue/scraper.py:220]: This is like black magic 😱 🤣

**Review by misialq — COMMENTED** [review]
In tests we trust!

I haven't tested on a Zotero library, but it's just adding some more IDs and the tests seem to cover it (see comment below) so that's probably ok.

**Review by lina-kim — APPROVED** [review]
Looks good and tests great :) Thanks @adamovanja!

File: q2_fondue/scraper.py (hunk 1 above)
Comment (adamovanja) [reply q2_fondue/scraper.py:380]: yes, that's right. thanks for the comment 👍🏼

File: q2_fondue/tests/test_scraper.py (hunk 2 above)
Comment (adamovanja) [reply q2_fondue/tests/test_scraper.py:245]: thanks for spotting this. It was actually not running correctly. I added and corrected it now. Let me know what you think.

Comment (misialq) [reply q2_fondue/tests/test_scraper.py:245]: Awesome, thanks!

**Pull request merged**